File: generator/src/test/resources/kotlin/expected/ProblemTypesTest/generates-problem-types.output2.kt
package io.test.service

import com.fasterxml.jackson.`annotation`.JsonIgnore
import java.net.URI
import kotlin.String
import org.zalando.problem.AbstractThrowableProblem
import org.zalando.problem.Exceptional
import org.zalando.problem.Status
import org.zalando.problem.ThrowableProblem

public class AccountNotFoundProblem(
  instance: URI? = null,
  cause: ThrowableProblem? = null,
) : AbstractThrowableProblem(TYPE_URI, "Account Not Found", Status.NOT_FOUND,
    "The requested account does not exist or you do not have permission to access it.", instance,
    cause) {
  @JsonIgnore
  override fun getCause(): Exceptional? = super.cause

  public companion object {
    public const val TYPE: String = "http://example.com/account_not_found"

    public val TYPE_URI: URI = URI(TYPE)
  }
}

File: generator/src/test/resources/swift/expected/ProblemTypesTest/generates-problem-types.output2.swift
import Foundation
import Sunday

public class AccountNotFoundProblem : Problem {

  public static let type: URL = URL(string: "http://example.com/account_not_found")!
  public override var description: String {
    return DescriptionBuilder(Self.self)
        .add(type, named: "type")
        .add(title, named: "title")
        .add(status, named: "status")
        .add(detail, named: "detail")
        .add(instance, named: "instance")
        .build()
  }

  public init(instance: URL? = nil) {
    super.init(type: Self.type, title: "Account Not Found", status: 404,
        detail: "The requested account does not exist or you do not have permission to access it.",
        instance: instance, parameters: nil)
  }

  public required init(from decoder: Decoder) throws {
    try super.init(from: decoder)
  }

  public override func encode(to encoder: Encoder) throws {
    try super.encode(to: encoder)
  }

}
